internal/gpt/gpt_test.go

```diff
@@ -0,0 +1,53 @@
+package gpt
+
+import (
+	"crypto/rand"
+	"testing"
+
+	efi "github.com/canonical/go-efilib"
+)
+
+func generatePartitionEntry(t *testing.T) *PartitionEntry {
+	var GUID efi.GUID = [16]byte{}
+	_, err := rand.Read(GUID[:])
+	if err != nil {
+		// Per rand.Read doc, this should never happen:
+		//    Read fills b with cryptographically secure random bytes.
+		//    It never returns an error, and always fills b entirely.
+		t.Fatal("failed to generate random UUID")
+	}
+
+	return &PartitionEntry{
+		PartitionEntry: &efi.PartitionEntry{
+			UniquePartitionGUID: GUID,
+		},
+	}
+}
+
+func TestFindByUUID(t *testing.T) {
+	partition1 := generatePartitionEntry(t)
+	partition2 := generatePartitionEntry(t)
+	partition3 := generatePartitionEntry(t)
+
+	table := []struct {
+		Name          string
+		UUIDStr       string
+		Parts         Partitions
+		ExpectedEntry *PartitionEntry
+	}{
+		{"symple", partition3.UniquePartitionGUID.String(), Partitions{partition1, partition2, partition3}, partition3},
```

Comment:
`simple` instead of `symple`, although I will suggest `match` instead of `simple` to reduce ambiguity